File: Tarea12.rb
=begin
Nombre:Jose Pablo González Flores Matr.:1986677 Hora:V5
=end
print("Números de la serie: ")
n= gets().to_i
a=2
b=1
par=1
c=-3
cont_n=1
cont_p=1
for i in(0...n).step 1
  if(cont_n>3) then
    b=1
    cont_n=1
  end
  if(cont_p>par) then
    b=2
    cont_p=1
    par=par+1
  end
  if(b==1) then
    print("#{a},")
    a=a+2
    cont_p = cont_p+1
  else
    print("#{c},")
    c=c-2
    cont_n=cont_n+1
  end
end

File: Tarea12.java
/* Nombre: JOSE PABLO GONZALEZ FLORES 
   Matr: 1986677 
   Hora: V5
 */
import javax.swing.JOptionPane;
public class Tarea12 {
    public static void main (String[] args){
 		String entrada;
		int a=1,b=2,d=-3,n,i=0,k;
		entrada=JOptionPane.showInputDialog("Numeros de la serie: ");
        n=Integer.parseInt(entrada);
        do{
            k=0;
		while(k<a&&i<n){
                    System.out.print(b+", ");
                    b=b+2;
                    k=k+1;
                    i=i+1;
    }
    a=a+1;
    if(i<n){
        for(int j=0;j<3;j++){
            System.out.print(d+", ");
            d=d-2;
            i=i+1;
            if(i>=n){
                break;
                 }
                  }
                   }
                    } while(i<n);
    }//Fin main

}//Fin de la clase

